Gems/Prefab/PrefabBuilder/PrefabGroup/PrefabGroupBehavior.cpp

```diff
@@ -309,7 +309,13 @@ namespace AZ::SceneAPI::Behaviors
                 AZStd::string meshGroupName = "default_";
                 meshGroupName += scene.GetName();
                 meshGroupName += meshNodePath;
-                AZ::StringFunc::Replace(meshGroupName, ".", "_");
+
+                // clean up the mesh group name
+                AZStd::replace_if(
```

Comment:
Nit: Is this called out somewhere that these names are invalid ie a mesh group name documentation? Could we link to that.

Also surprised theres no unit test/automated test change associated with this. Do we have a plan to add tests in this space? Seems like a prime candidate for unit tests.